lib/runtime.js

```diff
@@ -26,13 +27,18 @@ export function runtime(f) {
      */
     (
       /**
-       * @param {string | null} type
+       * @param {string | null | HFn} typeOrFn
        * @param {HProperties & {children?: HChild}} props
        * @returns {HResult}
        */
-      function (type, props) {
+      function (typeOrFn, props) {
         const {children, ...properties} = props
-        return type === null ? f(type, children) : f(type, properties, children)
+        if (typeof typeOrFn === 'function') {
+          return typeOrFn(props)
+        }
```

Comment:
Do this before the now potentially useless destructure: so above the `const ...`